### PR TITLE
add LOKI_URL env var for setting the URL

### PR DESCRIPTION
**Pull request by SvenDowideit**
more little things to help me try it out

### CODE REVIEW

**Comment by SvenDowideit** [comment]
feel free to not take my version :)

**Comment by SvenDowideit** [comment]
mmm, I guess I need a few more tuits:

```
 docker run -ti -v /dev/log:/dev/log -v /var/run/systemd/journal/:/var/run/systemd/journal/ --network loki_default svendowideit/journaldtail
ts="2019/01/11 04:45:40" msg="failed to tail journald: could not get next journal entry: finished wait but could not advance"
```

**Comment by SvenDowideit** [comment]
I get some weird things in Loki running:

```
docker run -ti -v /etc/machine-id:/etc/machine-id \
-v /dev/log:/dev/log \
-v /var/run/systemd/journal/:/var/run/systemd/journal/ \
-v /var/log/journal:/run/log/journal \
--network loki_default \
-e LOKI_URL=http://loki:3100/api/prom/push \
svendowideit/journaldtail
```

**Comment by SvenDowideit** [comment]
and eventually, it annoys loki:

```
loki_loki.1.tjiosskfzdpk@oa-30-mel    | level=warn ts=2019-01-11T05:03:44.810890187Z caller=gokit.go:43 msg="POST /api/prom/push (500) 1.037032ms Response: \"rpc error: code = Unknown desc = Entry out of order\\n\" ws: false; Accept-Encoding: gzip; Content-Length: 665; Content-Type: application/x-protobuf; User-Agent: Go-http-client/1.1; "
loki_loki.1.tjiosskfzdpk@oa-30-mel    | level=warn ts=2019-01-11T05:03:44.866320171Z caller=gokit.go:46 method=/logproto.Pusher/Push duration=79.421µs request="&PushRequest{Streams:[&Stream{Labels:{GLIB_DOMAIN=\"GoaBackend\", GLIB_OLD_LOG_API=\"1\", PRIORITY=\"4\", _AUDIT_LOGINUID=\"1000\", _AUDIT_SESSION=\"2\", _BOOT_ID=\"8a131ec51b27426e8cd0f6dcc14cf55c\", _CAP_EFFECTIVE=\"0\", _CMDLINE=\"/usr/lib/gnome-online-accounts/goa-daemon\", _COMM=\"goa-daemon\", _EXE=\"/usr/lib/gnome-online-accounts/goa-daemon\", _GID=\"1000\", _HOSTNAME=\"TOWER-SL\", _MACHINE_ID=\"2030bbd5a0634a65b4044b51db28cabe\", _PID=\"25118\", _SELINUX_CONTEXT=\"unconfined\\n\", _SOURCE_REALTIME_TIMESTAMP=\"1544429122157153\", _SYSTEMD_CGROUP=\"/user.slice/user-1000.slice/user@1000.service/dbus.service\", _SYSTEMD_INVOCATION_ID=\"6d759acf8d4a4ef2a5388aa54e5fa7f3\", _SYSTEMD_OWNER_UID=\"1000\", _SYSTEMD_SLICE=\"user-1000.slice\", _SYSTEMD_UNIT=\"user@1000.service\", _SYSTEMD_USER_SLICE=\"-.slice\", _SYSTEMD_USER_UNIT=\"dbus.service\", _TRANSPORT=\"journal\", _UID=\"1000\"},Entries:[{2018-12-10 08:05:22.157209 +0000 UTC goa_ews_client_autodiscover() failed: 2 — Error resolving “autodiscover.owa.csiro.au”: Name or service not known}],}],}" err="Entry out of order" msg="gRPC\n"
```

**Comment by SvenDowideit** [comment]
I _think_ what I then want to see (the Docker daemon logs) can be found with `{_COMM="dockerd"}` or `{_SYSTEMD_UNIT="docker.service"}` but atm, those only show from mid december last year, presumably because things go flop.

**Comment by SvenDowideit** [comment]
mmm, even running 

```
sudo journalctl --rotate
sudo journalctl --vacuum-time=2d
```

doesn't stop the Entry out of order issues - there are fewer of them though

**Comment by SvenDowideit** [comment]
interestingly, watching loki's the log output , it seems to only be having issues with entries from dockerd or kubelet

**Comment by SvenDowideit** [comment]
oh - and I presume atm, its intentionally exiting when there's no more log entries?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM       ubuntu:18.04
```
Comment (hikhvar) [inline Dockerfile]: I think it should be documented that the systemd c libraries must match in the log shipper image and the host system.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.PHONY: journaldtail docker
+
+docker: journaldtail
+	docker build -t $(HUBUSER)/journaldtail .
```
Comment (hikhvar) [inline Makefile:8]: Some default value for HUBUSER since this will trigger an error if not set.

**Review by hikhvar — CHANGES_REQUESTED** [review]
Two minor issues. I intend to merge this after they are resolved.

File: Dockerfile (hunk 1 above)
Comment (SvenDowideit) [reply Dockerfile]: oh, that's kinda aweful :/

Comment (SvenDowideit) [reply Dockerfile]: mmm, ok, in that case, I'll make this into a multi-stage build example that "works for me"

**Comment by hikhvar** [comment]
Thank you.

**Comment by SvenDowideit** [comment]
pleasure :) I'm not sure that i'll have time to do much with loki for a bit - its not quite at the level we need it to be to move, but your tool has been very helpful to my initial evaluation

**Pull request merged**